scipy/optimize/_direct.py

```diff
@@ -0,0 +1,182 @@
+r"""
+A python wrapper to the DIRECT algorithm implemented in Fortran
+===============================================================
+
+DIRECT is a method to solve global bound constraint optimization problems and
+was originally developed by D. R. Jones, C. D. Perttunen and B. E. Stuckmann.
+It is designed to find **global** solutions of mathematical optimization 
+problems of the from,
+
+.. math::
+
+       \min_ {x \in R^n} f(x)
+
+subject to
+
+.. math::
+
+       x_L \leq  x  \leq x_U
+
+Where :math:`x` are the optimization variables (with upper and lower
+bounds), :math:`f(x)` is the objective function.
+
+The DIRECT package uses the Fortran implementation of DIRECT written by
+Joerg.M.Gablonsky, DIRECT Version 2.0.4. More information on the DIRECT
+algorithm can be found in Gablonsky's `thesis 
+<http://repository.lib.ncsu.edu/ir/bitstream/1840.16/3920/1/etd.pdf>`_.
+
+Authors:
+
+Andreas Mayer <andimscience@gmail.com>, Amit Aides <amitibo@tx.technion.ac.il>
+"""
+
+import numpy as np
+from ._directmodule import direct
+from .optimize import OptimizeResult
+
+ERROR_MESSAGES = (
+    "u[i] < l[i] for some i",
+    "maxfun is too large",
+    "Initialization failed",
+    "There was an error in the creation of the sample points",
+    "An error occured while the function was sampled",
+    "Maximum number of levels has been reached.",
+)
+
+SUCCESS_MESSAGES = (
+    "Number of function evaluations done is larger then maxfun={}",
+    "Number of iterations is equal to maxiter={}",
+    ("The best function value found is within {} percent "
```

Comment:
Should we then remove `fglobal` parameter and always use the default? What do you say, @dschmitz89

Reply:
Imo setting it by default to ``-np.inf`` or ``None`` is a good choice. For benchmarking purposes, it can still be used. 

shgo also includes this option but it is called ``f_min``: https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.shgo.html

Maybe use that keyword then as well to make things more streamlined across optimize ?

Reply:
Addressed.

Reply:
N.B. the thread - https://github.com/scipy/scipy/pull/14300#discussion_r660435320